lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -923,9 +923,18 @@ ParseResult FIRParser::parseType(FIRRTLType &result, const Twine &message) {
           StringRef fieldName;
           FIRRTLType type;
           if (parseFieldId(fieldName, "expected bundle field name") ||
-              parseToken(FIRToken::colon, "expected ':' in bundle") ||
-              parseType(type, "expected bundle field type"))
+              parseToken(FIRToken::colon, "expected ':' in bundle"))
             return failure();
+          auto loc = getToken().getLoc();
+          if (parseType(type, "expected bundle field type"))
+            return failure();
+
+          // We require that elements of aggregates themselves
+          // support notion of FieldID, reject if the type does not.
+          if (!isa<hw::FieldIDTypeInterface>(type))
+            return emitError(loc, "bundle elements must support fieldID's, "
```

Comment:
I don't think the user is going to understand fieldIDs, maybe just leave it at "type 'blah' cannot be used as a field in a bundle"?

Reply:
You're totally right, I'll tighten this up, thank you!

(And 100% would be great to be able to render as FIRRTL!

That could be shared with the FIRRTL exporter, maybe, too?)